connectors/youtube.js

```diff
@@ -148,137 +154,110 @@ function cleanArtistTrack(artist, track) {
+    if (videoID && videoID.length > 0)
       videoID = videoID[1];
-   else
-      videoID = null;
-
-   // videoID from title at profile page
-   if ($('#playnav-curvideo-title > span[id!=chrome-scrobbler-status]').length > 0) {
-      videoID = $('#playnav-curvideo-title > span[id!=chrome-scrobbler-status]').attr('onclick').toString().match(/\?v=(.{11})/);
-      if (videoID && videoID.length > 0)
-         videoID = videoID[1];
-   }
-   // videoID from title at profile page - newer version
-   if ($('#playnav-curvideo-title > a').length > 0) {
-      videoID = $('#playnav-curvideo-title > a').attr('href').toString().match(/\?v=(.{11})/);
-      if (videoID && videoID.length > 0)
-         videoID = videoID[1];
-   }
+  }
+  // videoID from title at profile page - newer version
+  if ($('#playnav-curvideo-title > a').length > 0) {
+    videoID = $('#playnav-curvideo-title > a').attr('href').toString().match(/\?v=(.{11})/);
+    if (videoID && videoID.length > 0)
+      videoID = videoID[1];
+  }
 
-   // something changed?
-   if (!videoID) {
-      console.log('If there is a YouTube player on this page, it has not been recognized. Please fill in an issue at GitHub');
-      //alert('YouTube has probably changed its code. Please get newer version of the Last.fm Scrobbler extension');
-      return;
-   }
+  // something changed?
+  if (!videoID) {
+    console.log('If there is a YouTube player on this page, it has not been recognized. Please fill in an issue at GitHub');
+    //alert('YouTube has probably changed its code. Please get newer version of the Last.fm Scrobbler extension');
+    return;
+  }
+
+  var xhr = new XMLHttpRequest();
 
-   // http://code.google.com/intl/cs/apis/youtube/2.0/developers_guide_protocol_video_entries.html
-   var googleURL = "https://gdata.youtube.com/feeds/api/videos/" + videoID + "?alt=json";
-
-   // Get clip info from youtube api
-   chrome.runtime.sendMessage({type: "xhr", url: googleURL}, function(response) {
-      var info = JSON.parse(response.text);
-
-	  // Return early if not in music category
-      if (scrobbleMusicOnly) {
-		  var isMusic = false;
-		  for (var i = 0; i < info.entry.category.length; i++) {
-			  if (info.entry.category[i].term == 'Music') {
-				  isMusic = true;
-				  break;
-			  }
-		  }
-
-		  if (!isMusic) {
-			  console.log('Skipping track because it is not in Music category');
-			  return;
-		  }
-	  }
-
-      var parsedInfo = { artist: '', track: '' }; // FIXME: hotfix after YT API was shut down... parseInfo(info.entry.title.$t);
-      var artist = null;
-      var track = null;
-      var artist_dom = '';
-      var track_dom = '';
-      if (!feather) {
-        // Use the #eow-title #watch-headline-show-title if available
-        track_dom = $('#eow-title').clone();
-        artist_dom = $('#watch-headline-show-title', track_dom);
+  xhr.onreadystatechange = function () {
+    var info = null;
```

Comment:
Shouldn't we only continue if the readyState is 4?

e.g. add the following just below the function decoration?

```js
  if (xhr.readyState !== 4) {
  	return;
  }
```